fiftyone/utils/cvat.py

```diff
@@ -2377,6 +2425,2066 @@ def write(
+        if json is not None:
+            kwargs["json"] = json
+
+        response = self._session.post(**kwargs)
+        self._validate(response)
+        return response
+
+    def put(self, url, auth=None, data=None, files=None, json=None):
+        """Sends a PUT request to the given CVAT API URL.
+
+        Args:
+            url: the url
+            auth (None): authentication credentials
+            data (None): data to send
+            files (None): files to send
+            json (None): json data to send
+
+        Returns:
+            the request response
+        """
+        kwargs = {"url": url, "verify": False}
+        if auth is not None:
+            kwargs["auth"] = auth
+        if data is not None:
+            kwargs["data"] = data
+        if files is not None:
+            kwargs["files"] = files
+        if json is not None:
+            kwargs["json"] = json
+
+        response = self._session.put(**kwargs)
+        self._validate(response)
+        return response
+
+    def delete(self, url):
+        """Sends a DELETE request to the given CVAT API URL.
+
+        Args:
+            url: the url to send the request to
+
+        Returns:
+            the request response
+        """
+        response = self._session.delete(url, verify=False)
+        self._validate(response)
+        return response
+
+    def get_user_id(self, username):
+        """Retrieves the CVAT user ID for the given username.
+
+        Args:
+            username: the username
+
+        Returns:
+            the user ID, or None if the user was not found
+        """
+        if username is None:
+            return
+
+        if username in self._user_id_map:
+            return self._user_id_map[username]
+
+        search_url = self.user_search_url(username)
+        user_response = self.get(search_url)
+        resp_json = user_response.json()
+        for user_info in resp_json["results"]:
+            if user_info["username"] == username:
+                user_id = user_info["id"]
+                self._user_id_map[username] = user_id
+                return user_id
+
+        logger.warning("User '%s' not found in %s", username, search_url)
+
+    def create_task(
+        self,
+        labels=None,
+        segment_size=None,
+        image_quality=75,
+        task_assignee=None,
+        task_name="FiftyOne_annotation",
+    ):
+        """Creates a task on the CVAT server using the given label schema.
+
+        Args:
+            labels (None): the label schema to use for the created task
+            segment_size (None): maximum number of images to load into a job.
+                Not applicable to videos
+            image_quality (75): an int in `[0, 100]` determining the image
+                quality to upload to CVAT
+            task_assignee (None): the username to assign the created task(s)
+            task_name ("FiftyOne_annotation"): a name for the CVAT task
+
+        Returns:
+            a tuple of
```

Comment:
FYI I updated some of the `Returns:` sections here. Sphinx does not allow `Returns:` to have multiple "coloned" arguments like `Args:` does. You have to fiddle with it like you see here to make it look similar.

Generating the docs and inspecting how it looks is the easiest way I've found to learn what works. Inspecting existing code and copying its patterns is always good too.